torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -288,18 +303,42 @@ xla::XlaOp CreateIndexUpdate(
     dim_numbers.add_inserted_window_dims(i);
     dim_numbers.add_scatter_dims_to_operand_dims(i);
   }
-  // Build the combiner computation.
-  xla::XlaBuilder cb("scatter-combiner");
-  xla::Shape xla_scalar_shape =
-      xla::ShapeUtil::MakeShape(buffer_shape.element_type(), {});
-  xla::XlaOp p0 = xla::Parameter(&cb, 0, xla_scalar_shape, "p0");
-  xla::XlaOp p1 = xla::Parameter(&cb, 1, xla_scalar_shape, "p1");
-  if (combiner) {
-    combiner(p0, p1, &cb);
-  }
-  xla::XlaComputation combiner_computation = cb.Build().ConsumeValueOrDie();
+  xla::XlaComputation combiner_computation =
+      MakeScatterComputation(combiner, buffer_shape.element_type());
   return xla::Scatter(buffer, indices, new_values, combiner_computation,
                       dim_numbers);
 }
 
+xla::XlaOp CreateIndexFill(const xla::XlaOp& buffer, xla::int64 dim,
+                           const xla::XlaOp& index, const xla::XlaOp& value) {
+  xla::Shape buffer_shape = XlaHelpers::ShapeOfXlaOp(buffer);
+
+  xla::ScatterDimensionNumbers dim_numbers;
+  dim_numbers.set_index_vector_dim(1);
+  for (xla::int64 window_dim = 0; window_dim < buffer_shape.rank();
+       ++window_dim) {
+    if (window_dim != dim) {
+      dim_numbers.add_update_window_dims(window_dim);
+    } else {
+      dim_numbers.add_inserted_window_dims(window_dim);
+      dim_numbers.add_scatter_dims_to_operand_dims(dim);
```

Comment:
I'd use either dim or window_dim in both places here.

Reply:
Done.